### ISSUE TITLE
iOS autocorrect suggestions not supported

### ISSUE THREAD

**Issue by JonathanKolnik** [effektif/react-mentions#85]
I've gone in circles on this, I am not really sure how to solve it, but the insert is computed incorrectly, it should be ''. Also would it be possible to just replace the oldplaintext with the newplaintext entirely?

![image](https://cloud.githubusercontent.com/assets/1164060/13965697/bb78de3e-f045-11e5-827e-c9ecccb4cd2d.png)

Example: 
expected: ill -> I'll
actual: ill -> illl

utils.js
applyChangeToValue function 
plainTextValue -> I'll
selectionStartBeforeChange -> 3
selectionEndBeforeChange -> 3
selectionEndAfterChange -> 4
insert  -> 'l'

returns ill + l

**Comment by JonathanKolnik** [comment]
I attempted to modify handleChange in MentionInput to use the event.target.value if there was no mention, but then the next time something is mentioned it doesnt recognize it as you continue to type https://gyazo.com/aab6b63e24841aa52a4e1b0ce5e04a6e

https://gyazo.com/49b6746a7522235858270601d5c714dc

**Comment by jfschwarz** [comment]
Autocorrect unfortunately breaks the mentions input (see also the related issues for mobile FF #41 and Chrome #30)

I just pushed a new branch in an attempt to fix this. The implementation feels quite brittle but seems to pass the test case you outlined above. I would be very happy if you could test if this works in iOS Safari and maybe even contribute some more test cases.

I would be thrilled to get this merged, but atm don't really have time to properly test & finish this.

**Comment by JonathanKolnik** [comment]
Awesome, thank you! I will definitely test it out. As always, I really appreciate the speedy response and even more so the fix!

**Comment by JonathanKolnik** [comment]
Sorry, I tried to add this

 `    "react-mentions": "git://github.com/effektif/react-mentions.git#e3e5fb8b2248de88703f5ac567bf3bd085d9b8ca",
` 

to my package.json, which I think is the right way to use a specific branch/commit but all it installs is 
![image](https://cloud.githubusercontent.com/assets/1164060/13993400/9d3b3a8c-f0f6-11e5-9a4b-67aaca2c529f.png)

Any chance you know what I'm doing wrong?

**Comment by jfschwarz** [comment]
Hmm, this must have s.th. to do with the .npmignore as all the missing files are referenced there. I wasn't aware that this prevents installs from Github :unamused: 

I guess the quickest way around that, would be a clone and local install:

```
 git clone git@github.com:effektif/react-mentions.git
```

And then in your project:
```
npm i ../relative/path/to/react-mentions
```

**Comment by JonathanKolnik** [comment]
Hey, just checking up on this. Do you think you're close to releasing it? If not, would you mind describing briefly how I might build the package locally, so that I can drop that into my project. I think it is maybe a grunt task? I am struggling to install locally from the repo and I feel like I am missing a step. As always, I really appreciate the active maintenance of this repo and how helpful you've been!

**Comment by frontendphil** [comment]
Just released 0.4.2 which includes this fix. Please note the breaking changes, of 0.4.0
